### PR TITLE
Improve performance of the controlCharacterFilter method

### LINKED ISSUE #4208: HMDA Platform Processing Speed Improvements Part 2
After observing our current bottlenecks , improvements can be made to the HMDA Platform to increase our filling process speeds

- [ ] Updating Scala from 2.12 to 2.13 which gives us faster Futures and access to better number parsing
- [ ]  In edit checks, not parsing numbers if not necessary, and using the fast .toIntOption/.toDoubleOption instead of the exception-throwing variants

### PR DESCRIPTION
**Pull request by omnipresent07**
closes #4208

### CODE REVIEW

File: hmda/src/main/scala/hmda/validation/dsl/PredicateCommon.scala [hunk 1]
```diff
@@ -38,6 +38,8 @@ object PredicateCommon {
     Try(lessThanOrEqual(BigDecimal(upper)).check(BigDecimal(x))).getOrElse(false) && Try(greaterThanOrEqual(BigDecimal(lower)).check(BigDecimal(x))).getOrElse(false)
   }
 
+  def containedIn[A](domain: Set[A]): Predicate[A] = domain.contains(_: A)
+
   def containedIn[A](domain: Seq[A]): Predicate[A] = domain.contains(_: A)
```
Comment (lchen-2101) [inline hmda/src/main/scala/hmda/validation/dsl/PredicateCommon.scala:43]: are these 2 duplicated?

Comment (omnipresent07) [reply hmda/src/main/scala/hmda/validation/dsl/PredicateCommon.scala]: one uses `Seq` other uses `Set`. I added a commit to change the names

Comment (lchen-2101) [reply hmda/src/main/scala/hmda/validation/dsl/PredicateCommon.scala]: ah~ yeah, I see it now, I'm blind...

**Review by lchen-2101 — APPROVED** [review]
LGTM

**Pull request merged**